### PR TITLE
Feature/project setup and configuration

### PR DESCRIPTION
**Pull request by Lelik7777**
# Project Setup and Configuration Enhancements

This pull request introduces significant improvements to the project structure, development workflow, and build process. It sets up essential tools and configurations to ensure code quality, consistency, and efficient development practices.

### Code Quality and Formatting
- Added Stylelint and Prettier configurations for consistent code styling and formatting.
- Implemented Husky with pre-commit and pre-push hooks to enforce code quality checks.

### Project Structure
- Restructured directories for better organization, including a new 'ui' directory in components.
- Added index.ts files for improved module exports in button and select components.
- Removed unnecessary index.css and updated index.html accordingly.

### Styling
- Created a structured styles directory with separate files for colors, global styles, mixins, and typography.
- Integrated Inter font into the project and Storybook.

### Asset Management
- Added SVG icons and scripts to convert them into React components.
- Implemented scripts for renaming and updating SVG files.

### Build and Type Generation
- Updated vite-plugin-dts configuration to resolve build issues and improve type handling.
- Configured path aliases in tsconfig.json for better import readability.

### Dependency Management
- Moved sass and storybook from dependencies to devDependencies for proper categorization.
- Updated package.json with new scripts for various development tasks.

## Impact
These changes significantly improve the development experience, enforce coding standards, and enhance the overall project structure. The new configurations and tools will help maintain code quality and consistency across the project.

## Breaking Changes
- Developers will need to run `pnpm i` to install new dependencies and ensure all tools are properly set up.

## Next Steps
- Review and merge this PR.
- Ensure all team members are aware of the new development workflow and tools.
- Update documentation to reflect new project structure and development practices.

### CODE REVIEW

File: .prettierrc.cjs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  ...require('@it-incubator/prettier-config'),
+  "bracketSpacing": true,
+  //override settings here
+}
```
Comment (NoName-ForTeam) [inline .prettierrc.cjs:5]: all is okay

File: index.html [hunk 2]
```diff
@@ -8,6 +8,6 @@
   </head>
   <body>
     <div id="root"></div>
-    <script type="module" src="/src/main.tsx"></script>
+    <script type="module" src="/src/index.ts"></script>
   </body>
 </html>
```
Comment (NoName-ForTeam) [inline index.html:13]: must be <HtmL/>

File: .storybook/main.ts [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
   ],
   framework: {
-    name: "@storybook/react-vite",
+    name: '@storybook/react-vite',
     options: {},
   },
-};
-export default config;
+}
+export default config
```
Comment (NoName-ForTeam) [inline .storybook/main.ts:17]: export default {dich: 'diiiiiiiich'}

Comment (NoName-ForTeam) [inline .storybook/main.ts:17]: exxppxxpx

**Review by NoName-ForTeam — CHANGES_REQUESTED** [review]

File: .prettierrc.cjs (hunk 1 above)
Comment (Lelik7777) [reply .prettierrc.cjs:5]: ыфва

File: index.html (hunk 2 above)
Comment (Lelik7777) [reply index.html:13]: done

**Review by EvgeniyaYakunina — APPROVED** [review]

**Review by NoName-ForTeam — APPROVED** [review]

**Pull request merged**